backend/apps/core/services/base_service.py

```
@@ -0,0 +1,521 @@
+Provides abstract methods and utility functions to eliminate code duplication
+across service classes. All service classes should inherit from this base class.
+
+Addresses SOLID principles:
+- SRP: Separates concerns (validation, data access, business logic)
+- OCP: Extensible via validation rules and repository pattern
+- LSP: All services can be used interchangeably
+- ISP: Implements IServiceProtocol for consistent interface
+- DIP: Uses repository abstraction instead of direct model access
+"""
+from abc import ABC, abstractmethod
+from typing import Type, Optional, Dict, Any
```

Ruff (0.14.3)

`typing.Type` is deprecated, use `type` instead

(UP035)